app/services/responsibility_service.rb

```diff
@@ -182,24 +182,21 @@ def self.open_prison_nps_offender?(offender)
 
   class CrcRules
     def self.responsibility(offender)
-      if release_date_gt_12_weeks?(offender)
+      # CRC can look at HDC date, NPS is not supposed to
+      earliest_release_date =
+        offender.home_detention_curfew_actual_date.presence ||
+          [offender.automatic_release_date,
+           offender.conditional_release_date,
+           offender.home_detention_curfew_eligibility_date].compact.min
+
+      return nil if earliest_release_date.nil?
```

Comment:
In the case that we return `nil`, what does this ultimately mean – is it that the POM is not responsible because the COM is?

I wonder whether it would be clearer/more descriptive to have this case as another class constant, alongside the `RESPONSIBLE`, `SUPPORTING`, `COWORKING` and `NPS` constants defined on lines 10-13.

If this represents what I think it does, then I guess it could be a constant called `NOT_RESPONSIBLE` or `COM_RESPONSIBLE`?

I also wonder whether the return value itself should/could be a new `Responsiblity` object with a `:custody?` value of `false`, rather than the implicit `false` garnered from the `nil` response.

I think this could make this code a little more explicit in what it's trying to calculate. What do you think @starswan?

Reply:
This module got changed at some point to allow a return of nil for 'dont know' which looks like a receipe for disaster (I think the null object pattern might have been a better choice). I didn't address this challenge in this PR.